src/feature/relay/selftest.c

```diff
@@ -272,7 +272,7 @@ router_do_reachability_checks(int test_or, int test_dir)
 int
 inform_testing_reachability(void)
```

Comment:
It would probably be better for this function to take as an argument which ports it is testing.  Otherwise the message it sends risks getting out of sync with the tests that were actually launched.

(That is to say, this function claims that an IPv6 orport check was launched when an IPv6 address exists -- not directly when an IPv6 orport check was launched.)

Reply:
I agree in principle.

But we don't report that we're testing reachability until at least one circuit has finished extending. (That circuit can be any purpose, not just testing reachability.)

So there are a bunch of subtle constraints and race conditions here. I split this change into another ticket:
https://trac.torproject.org/projects/tor/ticket/34137

And documented the issues in the function, see 492c512af4a4f9ee143b945296fd1c61fc68dba1.